Added enumeration datatypes to the spec (#75)
AllenInstitute/sonata: docs/SONATA_DEVELOPER_GUIDE.md

For `biophysical` models defined according to the *hoc* (see above), the namespace definition is to be filled in by the Allen folks.

#### Nodes - Enum datatypes

> It is often the case that attributes storing strings have a limited number of possible values (for instance, `model_type`, or cell morphological type).

Reviewer: "attributes storing strings" -> "string attributes" or "attributes of string type"

Author: Done.

> For space efficiency, these attributes are better be represented with *enumerations*, i.e. data types consisting of a limited set of named integer values.

Reviewer: ```suggestion
For space efficiency, it is better to represent these attributes with *enumerations*, i.e. data types consisting of a limited set of named integer values.
```

> However, Enum Datatypes offered by HDF5 impose a limit on the total number of possible values due to 64K limit on object header limitation:</br>

Reviewer: Use Markdown syntax for this link
```suggestion
However, Enum Datatypes offered by HDF5 impose a limit on the total number of possible values due to [64K limit on object header limitation](https://support.hdfgroup.org/HDF5/hdf5-quest.html).
```

https://support.hdfgroup.org/HDF5/hdf5-quest.html

> To work around this limitation, we support *explicit enumerations* in SONATA Nodes / Edges HDF5 files.

Reviewer: ```suggestion
 To work around this limitation, SONATA nodes/edges HDF5 files may use *explicit enumerations*.
```

Each attribute `/<population>/<group>/X` with integer datatype may have a corresponding attribute `/<population>/<group>/@library/X` with a limited set of string values. In this case `X` should be interpreted as an index into `@library/X`; and querying `X` values should return corresponding string values instead of integers.

> HDF5 group `@library` is reserved for this purpose.

Reviewer: ```suggestion
The group `@library` is reserved for this purpose.
```

Reviewer: > querying `X` values should return corresponding string values instead of integers.

I think this sentence is not necessary. An implementer may choose to do this or keep the integers, requiring clients of the API to resolve the indices to strings manually (this is the case in Brain, and I'm pretty sure we will keep it like that for efficiency)
